### PR TITLE
Update Senior 2 description to be more accurate

### PR DESCRIPTION
**Pull request by lukemorton**
## What

- Explain that Senior 2 role is varied and challenging
- Ensure we are clear that we aren't looking for unicorns but looking for people who would enjoy a broad technology role even if they don't already have that experience
- Be clear that we require customer and/or consulting experience

## Why

Our Senior 2 role is rather outdated. Hopefully this improves things.

### CODE REVIEW

File: roles/senior_software_engineer_2.md [hunk 1]
```diff
@@ -48,9 +48,9 @@ Don’t forget to mention any of the experience listed below. While it’s optio
 - Consultancy
 - Pair programming – we pair around 50% of the time
 - Component-based design techniques such as using pattern libraries, styled-components, CSS-in-JS, BEM, and/or SUIT CSS
-- Debugging infrastructure
+- Debugging infrastructure (AWS/Heroku/Linux/HTTP, GCP and Azure)
```
Comment (StevenLeighton21) [inline roles/senior_software_engineer_2.md]: Isn't this a revert from the last PR?

File: roles/senior_software_engineer_2.md [hunk 2]
```diff
@@ -48,9 +48,9 @@ Don’t forget to mention any of the experience listed below. While it’s optio
...
 - The React ecosystem including a test-driven approach
-- Infrastructure as code technology like Terraform and Cloud Formation
+- Infrastructure as code tools such as Ansible, Chef, Puppet, Terraform and/or Saltstack
```
Comment (StevenLeighton21) [inline roles/senior_software_engineer_2.md]: Same comment: is this reverting the last PR?

File: roles/senior_software_engineer_2.md [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 
-We primarily write and deliver custom software to our customers. Before a project kicks off we will have a number of sessions where we begin to understand the customer’s business and what they are looking to achieve. We then formulate a roadmap for them, potentially a business case too, before starting delivery.
-
 We primarily write and deliver custom software for the public sector. We work across central and local government, as well as in health, and our past lies in the technology startup world. Technical excellence for us isn’t about delivering to feature lists. We place a strong emphasis on outcome-based delivery; ensuring our customer’s goals are understood and achieved with the technology we deploy.
 
+Depending on our customers' needs our Senior 2s may play one or more role. Sometimes the role is very hands-on and you'll be the most senior engineer on the team delivering software, other times you will play more of a technical architect advising teams and ensuring consistency across multiple deliveries. Further still you may play a strong part in account management and hold relationships with our customers senior staff as a technical adviser.
```
Comment (kjdchapman) [inline roles/senior_software_engineer_2.md]: nit:typo 
```suggestion
Depending on our customers' needs our Senior 2s may play one or more roles. Sometimes the role is very hands-on and you'll be the most senior engineer on the team delivering software, other times you will play more of a technical architect advising teams and ensuring consistency across multiple deliveries. Further still you may play a strong part in account management and hold relationships with our customers senior staff as a technical adviser.
```

File: roles/senior_software_engineer_2.md [hunk 4]
```diff
@@ -24,8 +26,6 @@ We grow a team of polyglot programmers, which you might already consider yoursel
 
 ## What experience are we looking for?
 
-While we will look for you to have experience in these things, if you don’t have one of these don’t let that stop you applying.
```
Comment (kjdchapman) [inline roles/senior_software_engineer_2.md]: I'm in favour of keeping this paragraph, or something like it. Less represented groups are proportionately less likely to apply if they don't have 100% of the "requirements" for a job.

File: roles/senior_software_engineer_2.md (hunk 1 above)
Comment (lukemorton) [reply roles/senior_software_engineer_2.md]: Fixed

File: roles/senior_software_engineer_2.md (hunk 2 above)
Comment (lukemorton) [reply roles/senior_software_engineer_2.md]: Fixed

File: roles/senior_software_engineer_2.md (hunk 4 above)
Comment (lukemorton) [reply roles/senior_software_engineer_2.md]: It's a mistake, will add back.

File: roles/senior_software_engineer_2.md [hunk 5]
```diff
@@ -1,15 +1,17 @@
 # Senior Software Engineer 2
 
-Manchester & London, UK. Often advertised as Lead Software Engineer.
+(Often advertised as Lead Software Engineer.)
 
-Our Senior Engineer 2s turn our customer’s needs into roadmaps, coach teams to deliver, are cutting the best code of their career, still learn every day, and constantly strive to be nice humans.
+Our Senior Engineer 2s are Tech Leads, Architects, Thought Leaders, Full Stack Engineers, and Consultants. They are our most senior technologist within a customer account representing Made Tech and act as a hands-on CTO across one or more software delivery teams. They combine technical excellence, drive to deliver, and coaching, to achieve outcomes for our customers and their users, and to establish strong engineering cultures within our customers organisations. They find themselves working on a variety of different problems from monoliths to microservices, upskilling colleagues and customers, always finding themselves learning from others, while constantly striving to be nice humans :)
```
Comment (sarrionandia) [inline roles/senior_software_engineer_2.md]: I think more emphasis on outcomes would be good here, IMO we're pretty good at measuring Senior 2s by how they are achieving business outcomes as opposed to by what specific tasks they are carrying out.

File: roles/senior_software_engineer_2.md [hunk 6]
```diff
@@ -1,15 +1,17 @@
...
-
 We primarily write and deliver custom software for the public sector. We work across central and local government, as well as in health, and our past lies in the technology startup world. Technical excellence for us isn’t about delivering to feature lists. We place a strong emphasis on outcome-based delivery; ensuring our customer’s goals are understood and achieved with the technology we deploy.
 
+Depending on our customers' needs our Senior 2s may play one or more roles. Sometimes the role is very hands-on and you'll be the most senior engineer on the team delivering software, other times you will play more of a technical architect advising teams and ensuring consistency across multiple deliveries. Further still you may play a strong part in account management and hold relationships with our customers senior staff as a technical adviser.
+
+We need flexible individuals who enjoy a varied workload and want to switch between being more and less hands-on. It might be that you're stronger as a Tech Lead or as a Tech Architect, or only have experience in one area, that's okay as long as you have an interest and willingness to broaden your expertise.
```
Comment (sarrionandia) [inline roles/senior_software_engineer_2.md]: Can we make it clear that we aren't really about being 'post-technical' though? Some roles might be architect-y, but in general you probably wouldn't get on at made tech if you think coding is something you just did while you were more junior. That's probably the main reason I've rejected experienced candidates in the past.

Comment (lukemorton) [reply roles/senior_software_engineer_2.md]: Yeah great point!

Comment (lukemorton) [reply roles/senior_software_engineer_2.md]: How about this?

"We need flexible individuals who enjoy a varied workload, are happy to switch between being more and less hands-on, but don't see themselves stepping away from code any time soon."

**Pull request merged**